app/helpers/patient_query_helper.rb

```diff
@@ -511,23 +511,35 @@ def advanced_filter_lab_result(patients, filter)
     filter[:value].each do |field|
       case field[:name]
       when 'lab-type'
-        labs = labs.where('laboratories.lab_type = ?', field[:value])
+        labs = if field[:value].blank?
+                 labs.where('laboratories.lab_type = ?', '').or(labs.where('laboratories.lab_type IS NULL'))
```

Comment:
A nicer, more "ActiveRecord" way to handle some of these is (example for this line):
```
labs.where(laboratories: { lab_type: ['', nil] })
```

Reply:
ah, this is the syntax I was looking for. addressed here: https://github.com/SaraAlert/SaraAlert/pull/961/commits/e1593558484e9244342076ad670f9a75b6bc4e5c

@hackrm the parenthesis was removed in this commit as well